applications/ShallowWaterApplication/custom_processes/id_renumbering_process.cpp

```diff
@@ -0,0 +1,177 @@
+    IndexType unique_id = 0;
+    mOriginElementsIds.clear();
+
+    // Loop the model parts
+    for (const auto& name : mModelPartNames)
+    {
+        ModelPart& model_part = mrModel.GetModelPart(name);
+        for (int i = 0; i < static_cast<int>(model_part.NumberOfElements()); ++i)
+        {
+            auto it_elem = model_part.ElementsBegin() + i;
+            mOriginElementsIds[++unique_id] = it_elem->Id();
+            it_elem->SetId(unique_id);
+        }
+    }
+}
+
+void IdRenumberingProcess::RenumberConditions()
+{
+    // The new absolute unique_id
+    IndexType unique_id = 0;
+    mOriginConditionsIds.clear();
+
+    // Loop the model parts
+    for (const auto& name : mModelPartNames)
+    {
+        ModelPart& model_part = mrModel.GetModelPart(name);
+        for (int i = 0; i < static_cast<int>(model_part.NumberOfConditions()); ++i)
+        {
+            auto it_cond = model_part.ConditionsBegin() + i;
+            mOriginConditionsIds[++unique_id] = it_cond->Id();
+            it_cond->SetId(unique_id);
+        }
+    }
+}
+
+void IdRenumberingProcess::RestoreNodes()
+{
+    if(mOriginNodesIds.size() != 0)
+    {
+        // Loop the model parts
+        for (const auto& name : mModelPartNames)
+        {
+            ModelPart& model_part = mrModel.GetModelPart(name);
+            for (int i = 0; i < static_cast<int>(model_part.NumberOfNodes()); ++i)
+            {
+                auto it_node = model_part.NodesBegin() + i;
+                it_node->SetId(mOriginNodesIds[it_node->Id()]);
+            }
+        }
+    }
+    // Avoiding the possibility to restore the ids twice
+    mOriginNodesIds.clear();
+}
+
+void IdRenumberingProcess::RestoreElements()
+{
+    if (mOriginElementsIds.size() != 0)
+    {
+        // Loop the model parts
+        for (const auto& name : mModelPartNames)
+        {
+            ModelPart& model_part = mrModel.GetModelPart(name);
+            for (int i = 0; i < static_cast<int>(model_part.NumberOfElements()); ++i)
+            {
+                auto it_elem = model_part.ElementsBegin() + i;
+                it_elem->SetId(mOriginElementsIds[it_elem->Id()]);
+            }
+        }
+    }
+    // Avoiding the possibility to restore the ids twice
+    mOriginElementsIds.clear();
+}
+
+void IdRenumberingProcess::RestoreConditions()
+{
+    if (mOriginConditionsIds.size() != 0)
+    {
+        // Loop the model parts
+        for (const auto& name : mModelPartNames)
+        {
+            ModelPart& model_part = mrModel.GetModelPart(name);
+            for (int i = 0; i < static_cast<int>(model_part.NumberOfConditions()); ++i)
+            {
+                auto it_cond = model_part.ConditionsBegin() + i;
+                it_cond->SetId(mOriginConditionsIds[it_cond->Id()]);
+            }
+        }
+    }
+    // Avoiding the possibility to restore the ids twice
+    mOriginConditionsIds.clear();
+}
+
+StringVectorType IdRenumberingProcess::GetRootModelPartNames() const
```

Comment:
I have some concerns regarding this method: 
1) Why you can't use the `GetRootModelPart()` method with each one of the model parts of the model?
2) You are iterating all the model parts in the model container. What happens if the model contains other model parts that are not related with the "shallow waters" problem?
3) If you need this method because you can't use the `GetRootModelPart()`, does it work with recursive model parts (e.g. ModelPartA.SubModelPartB.SubSubModelPartC)?

Reply:
1. Now I'm using `GetRootModelPart()`
2. There are two constructors: one with a vector of strings to specify the "shallow water" related model parts. The other constructor calls this method and assumes that all model parts are related to the "shallow water" problem.
3. Solved with 1